.github/workflows/tests.yml

```diff
@@ -42,6 +46,10 @@ jobs:
       - name: Build
         uses: ./.github/workflows/actions/build
 
+      - name: Type declarations (optional)
+        if: github.event.inputs.types == 'true'
+        run: npm run build:types
```

Comment:
**question**: Because the step is in the `Build` job, that'll mark that the build failed, which sounds pretty alarming for an optional check. It will muddy things between bundling having failed or type checks having failed until you dig in the job logs. Does it necessarily have to be grouped with the `Build` (for performance, maybe? or was it just in anticipation that we'd use this to build a `.d.ts` file at some point in the future)? Could it be an entry in the linting matrix instead to highlight that it's specifically the types that failed in the list of checks?

Reply:
Yeah spot on! I did this on purpose

Do I need to rename the word "optional"?

**Don't tick it:** Types can't break the build
**Do tick it:** Types can break the build

In terms of grouping it with the **Build** step, it's because the TypeScript CLI command is `tsc --build` and would (eventually) output our type definitions. It's always skipped until you tick the box.

### Lint task matrix
Sadly it's not possible for a job-level `if:` to conditionally check values from "inputs" against matrix "include" context without using [`${{ fromJSON() }` to build the matrix dynamically](https://stackoverflow.com/questions/65384420/how-do-i-make-a-github-action-matrix-element-conditional) 😱 

The alternative is to drop to step-level `if:` but only EVERY matrix step 😬

```yaml
if: ${{ matrix.name !== 'lint-types' || github.event.inputs.types == 'true' }}
```

Which gets pretty repetitive:
```yaml
steps:
  - name: Checkout
    if: ${{ matrix.name !== 'lint-types' || github.event.inputs.types == 'true' }}
    uses: actions/checkout@v3

  - name: Restore dependencies
    if: ${{ matrix.name !== 'lint-types' || github.event.inputs.types == 'true' }}
    uses: ./.github/workflows/actions/install-node

  - name: Cache linter
    if: ${{ (matrix.name !== 'lint-types' || github.event.inputs.types == 'true') && matrix.cache }}
    uses: actions/cache@v3
    with:
      key: ${{ matrix.name }}-cache-${{ runner.os }}
      path: ${{ matrix.cache }}

  - name: Run lint task
    if: ${{ matrix.name !== 'lint-types' || github.event.inputs.types == 'true' }}
    run: ${{ matrix.run }}
```